server/util/cookie/cookie.go

```diff
@@ -23,10 +26,19 @@ var (
 	httpsOnlyCookies = flag.Bool("auth.https_only_cookies", false, "If true, cookies will only be set over https connections.")
 )
 
+func cookieDomain() string {
+	domain := ""
+	if subdomain.Enabled() {
+		domain = build_buddy_url.Domain()
+	}
+	return domain
+}
```

Comment:
When switching from one domain to another, I wonder if we should detect cookies from the old domain (`app.buildbuddy.dev` / `app.buildbuddy.io`) and ignore/delete those? See https://buildbuddy-corp.slack.com/archives/C01D5GHRJ59/p1692108833397279

Reply:
Updated PR to clear non-domain bound cookies when setting domain bound cookies.